src/server/game/Maps/MapInstanced.cpp

```diff
@@ -163,7 +163,11 @@ Map* MapInstanced::CreateInstanceForPlayer(const uint32 mapId, Player* player, u
             {
                 groupBind = group->GetBoundInstance(this);
                 if (groupBind)
+                {
+                    // solo saves should be reset when entering a group's instance
+                    player->UnbindInstance(GetId(), player->GetDifficulty(IsRaid()));
```

Comment:
Where is MapInstanced::CreateInstanceForPlayer called from? UnbindInstance is not threadsafe